shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -569,6 +571,22 @@ - (void)blockGesture {
     case FlutterPlatformViewGestureRecognizersBlockingPolicyEager:
       // We block all other gesture recognizers immediately in this policy.
       self.delayingRecognizer.state = UIGestureRecognizerStateEnded;
+
+      // On iOS 18.2, WKWebView's internal recognizer likely caches the old state of its blocking
+      // recognizers (i.e. delaying recognizer), resulting in non-tappable links. See
+      // https://github.com/flutter/flutter/issues/158961. Removing and adding back the delaying
+      // recognizer solves the problem, possibly because UIKit notifies all the recognizers related
+      // to (blocking or blocked by) this recognizer. It is not possible to inject this workaround
+      // from the web view plugin level. Right now we only observe this issue for
+      // FlutterPlatformViewGestureRecognizersBlockingPolicyEager, but we should try it if a similar
+      // issue arises for the other policy.
+      if (@available(iOS 18.2, *)) {
+        if ([self.embeddedView isKindOfClass:[WKWebView class]]) {
```

Comment:
@jmagman @cbracken FYI I limited the workaround to web view, as discussed here: https://github.com/flutter/flutter/issues/158961#issuecomment-2518741174

Reply:
That sgtm. If we discover that this is more broadly applicable, it's well-documented here and someone can drop the conditional.